test/listen.test.js

```diff
@@ -186,11 +186,13 @@ test('listen twice on the same port callback with (err, address)', t => {
 // https://nodejs.org/api/net.html#net_ipc_support
 if (os.platform() !== 'win32') {
   test('listen on socket', t => {
-    t.plan(3)
+    t.plan(4)
     const fastify = Fastify()
-    t.tearDown(fastify.close.bind(fastify))
+    const socketFileName = Date.now().toString().concat('.sock')
+    const sockFile = path.join(os.tmpdir(), socketFileName)
+
+    t.tearDown(fastify.close.bind(t.equal(fs.existsSync(sockFile), false)))
```

Comment:
Can you remove the bind? I think you are executing the `t.equal` too early (at bind time), and not when we are tearing down the test.

Reply:
If I use something like 

`t.tearDown(() => fastify.close(() => t.equal(fs.existsSync(sockFile), false)))`

I receive this error

![image](https://user-images.githubusercontent.com/9164563/61132567-eb98a580-a4bb-11e9-823d-f1555936ad63.png)

Reply:
Alternatively I can simply do something like

```
t.tearDown(() => fastify.close(() => {
  if (fs.existsSync(sockFile, true)) {
    fs.unlinkSync(sockFile)
  }
}))
```

Reply:
I think the case could be due to a silent error in the `try/catch` block and doing so will run twice this file removal

@allevo do you know why there is a sync unlink just before the `listen` call?
Could we move it in the `teardown` function?